pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -1369,26 +1318,13 @@ func (c *Reconciler) updatePipelineRunStatusFromInformer(ctx context.Context, pr
 }
 
 func updatePipelineRunStatusFromChildObjects(ctx context.Context, logger *zap.SugaredLogger, pr *v1beta1.PipelineRun, taskRuns []*v1beta1.TaskRun, runObjects []v1beta1.RunObject) error {
-	cfg := config.FromContextOrDefaults(ctx)
-
-	switch cfg.FeatureFlags.EmbeddedStatus {
-	case config.FullEmbeddedStatus:
-		// Clear any ChildReferences that are present.
-		// This can occur if the value of "embedded-status" was modified during PipelineRun execution.
-		pr.Status.ChildReferences = nil
-		updatePipelineRunStatusFromTaskRuns(logger, pr, taskRuns)
-		updatePipelineRunStatusFromCustomRunsOrRuns(logger, pr, runObjects)
-	case config.BothEmbeddedStatus:
-		updatePipelineRunStatusFromTaskRuns(logger, pr, taskRuns)
-		updatePipelineRunStatusFromCustomRunsOrRuns(logger, pr, runObjects)
-		updatePipelineRunStatusFromChildRefs(logger, pr, taskRuns, runObjects)
-	default:
-		// Clear any TaskRuns and Runs present in the status.
-		// This can occur if the value of "embedded-status" was modified during PipelineRun execution.
-		pr.Status.Runs = nil
-		pr.Status.TaskRuns = nil
-		updatePipelineRunStatusFromChildRefs(logger, pr, taskRuns, runObjects)
-	}
+	// Clear any TaskRuns and Runs present in the status.
+	// This can occur if the value of "embedded-status" was modified during PipelineRun execution prior to its removal.
```

Comment:
Is this still necessary without an embedded-status flag?

Reply:
I was trying to be safe earlier to not to delete the guards before removing the `status.taskruns` or `status.runs` from the apis and also to create the cleanup issue as TODO. I am happy to clean them up if we are doing so in this release.

Reply:
I think this could still be valuable before we remove the `status.runs` `status.taskruns` corresponding to the test case we might want at https://github.com/tektoncd/pipeline/blob/main/pkg/reconciler/pipelinerun/pipelinerun_updatestatus_test.go#L1028-L1058?

Reply:
I think we should remove or rewrite test cases with taskruns/runs